code/__HELPERS/icon_smoothing.dm

```diff
@@ -224,43 +233,36 @@
 		else if(adjacencies & N_EAST)
 			se = "4-e"
 
+	var/list/New = list()
+
 	if(A.top_left_corner != nw)
 		A.overlays -= A.top_left_corner
 		A.top_left_corner = nw
-		A.overlays += nw
+		New += nw
 
 	if(A.top_right_corner != ne)
 		A.overlays -= A.top_right_corner
 		A.top_right_corner = ne
-		A.overlays += ne
+		New += ne
 
 	if(A.bottom_right_corner != sw)
 		A.overlays -= A.bottom_right_corner
 		A.bottom_right_corner = sw
-		A.overlays += sw
+		New += sw
 
 	if(A.bottom_left_corner != se)
 		A.overlays -= A.bottom_left_corner
 		A.bottom_left_corner = se
-		A.overlays += se
-
-/proc/find_type_in_direction(atom/source, direction, range=1)
-	var/x_offset = 0
-	var/y_offset = 0
-
-	if(direction & NORTH)
-		y_offset = range
-	else if(direction & SOUTH)
-		y_offset -= range
+		New += se
 
-	if(direction & EAST)
-		x_offset = range
-	else if(direction & WEST)
-		x_offset -= range
+	if(New.len)
+		A.overlays.Add(New)
```

Comment:
Would this normally use `add_overlays(New)` on /tg/? Because these two things are not 1:1 with each other.

Reply:
This doesn't normally use add_overlays on TG, no.

Reply:
This does use add_overlays on /tg/...